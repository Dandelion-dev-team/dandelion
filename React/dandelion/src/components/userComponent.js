import { Link } from "gatsby"
import React, { useEffect, useState } from "react"
import "../styles/App.scss"

export default function CrudComponent(props) {
  const [users, setUsers] = useState(0);

  useEffect(() => {
    // Update the document title using the browser API
    fetch("http://localhost:3000/users", {
      method: "GET",
      headers: new Headers({
        'Cache-Control': 'no-cache, no-store, must-revalidate',
        'Pragma': 'no-cache',
        'Expires': 0,
      })
    }).then(response => response.json())
      .then(
        data => setUsers(data))
  }, []);



  const deleteUser = index => {
    fetch("http://localhost:3000/users/" + index, {
      method: "DELETE",
      headers: { 'Content-Type': 'application/json' },
    }).then(console.log("delete " + index)).then(window.location.reload(false))
  }

  const editUser = (user) => {
    props.parentCallback(user);
  }

  return (
    <div className="authTable">
      <table className="tableList">
        <thead>
          <tr>
            <th>ID</th>
            <th>School ID</th>
            <th>Username</th>
            <th>Superuser</th>
            <th>Status</th>
            <th>Edit</th>
            <th>Delete</th>
          </tr>
        </thead>

        {users ? users.map(user => (

          <tbody key={users.id}>
            <td>{user.id}</td>
            <td>{user.school_id}</td>
            <td>{user.username}</td>
            <td>{user.is_superuser.toString()}</td>
            <td>{user.status}</td>
            <td>
              <div className="submit-btn">
                <input
                  type="submit"
                  className="submitButton"
                  value="Edit"
                  onClick={() => { editUser(user) }}
                ></input>
              </div>
            </td>
            <td>
              <input
                type="submit"
                className="submitButton"
                value="Delete"
                onClick={() => { deleteUser(user.id) }}
              ></input>
            </td>
          </tbody>
        )) : null}

      </table>
    </div>
  )
}