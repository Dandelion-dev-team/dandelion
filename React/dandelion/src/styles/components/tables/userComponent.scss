.userTable{

    .tableList {
        margin-top: 1.5rem;
        margin-left: 2rem;
      tbody {
          
         &:hover {
             background-color: #f7f8ff;
             cursor: pointer;
         }
      }
        
    }

   
    
}