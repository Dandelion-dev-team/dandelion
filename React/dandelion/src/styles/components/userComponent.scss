.authTable{
    margin-left: 100px;

    tr td {
        display:block;
        width:100%;
    }
}