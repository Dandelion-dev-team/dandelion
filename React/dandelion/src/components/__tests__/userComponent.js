import React from "react"
import renderer from "react-test-renderer"
import userComponent from "../userComponent"


describe("Tags", () => {
    let originalFetch;

    beforeEach(() => {
        originalFetch = global.fetch;
        global.fetch = jest.fn(() => Promise.resolve({
            json: () => Promise.resolve(
                {
                    "id": 1,
                    "school_id": "GLA11",
                    "username": "newnewname",
                    "password_hash": "newpassword",
                    "is_sysadmin": true,
                    "is_superuser": true,
                    "status": "active",
                    "notes": "Took away admin powers "
                },
                {
                    "id": 3,
                    "school_id": "GLA11",
                    "username": "brian",
                    "password_hash": "asd",
                    "is_sysadmin": false,
                    "is_superuser": true,
                    "status": "active",
                    "notes": " "
                }
            )
        }));
    });

    afterEach(() => {
        global.fetch = originalFetch;
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(<userComponent />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})

