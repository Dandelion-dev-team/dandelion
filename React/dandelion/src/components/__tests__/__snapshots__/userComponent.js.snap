// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly 1`] = `<userComponent />`;
